pkg/api/v1beta3/types.go

```diff
@@ -781,7 +781,7 @@ type PodSpec struct {
 	// Optional: Set DNS policy.  Defaults to "ClusterFirst"
 	DNSPolicy DNSPolicy `json:"dnsPolicy,omitempty" description:"DNS policy for containers within the pod; one of 'ClusterFirst' or 'Default'"`
 	// NodeSelector is a selector which must be true for the pod to fit on a node
-	NodeSelector map[string]string `json:"nodeSelector,omitempty" description:"selector which must match a node's labels for the pod to be scheduled on that node"`
+	NodeSelector string `json:"nodeSelector,omitempty" description:"selector which must match a node's labels for the pod to be scheduled on that node"`
```

Comment:
We can't break v1beta3 now. We just asked everyone to migrate to it.

We still have the opportunity to make breaking changes in the v1 API #7018, but should do so carefully.

I'll take a look at the PR. I'm not sure how I feel about representing the selectors as strings in the API.